doc/keymap-format-text-v1.md

```diff
@@ -829,42 +864,389 @@ being considered when matching the modifier state against the type.
+
+    // Define level names
+    level_name[Level1] = "Base";
+    level_name[Level2] = "Shift";
+};
+```
+            @endfigcaption
+            @figure@figcaption
+            Mapping test
+            @endfigcaption
+| *Active* modifiers | *Filtered* modifiers | Match? | Shift level |
+| ------------------ | -------------------- | ------ | ----------- |
+| (none)             | (none)               | Yes    | 1           |
+| `Shift`            | `Shift`              | Yes    | 2           |
+| `Lock`             | (none)               | Yes    | 1           |
+| `Shift + Lock`     | `Shift`              | Yes    | 2           |
+            @endfigcaption
+        </div>
+    </div>
+    <div class="example">
+        <div class="example-inner">
+            <div class="example-title">`ALPHABETIC` @anchor ALPHABETIC</div>
+            @figure@figcaption
+            Definition code ([source][alphabetic-type-src])
+            @endfigcaption
+```c
+type "ALPHABETIC" {
+    // Only care about Shift and Lock
+    modifiers = Shift + Lock;
+    // Define mapping
+    map[None]  = Level1; // No modifier   -> level 1
+    map[Shift] = Level2; // Exactly Shift -> level 2
+    map[Lock]  = Level2; // Exactly Lock  -> level 2
+    // Define level names
+    level_name[Level1] = "Base";
+    level_name[Level2] = "Caps";
+};
+```
+            @endfigcaption
+            @figure@figcaption
+            Mapping test
+            @endfigcaption
+| *Active* modifiers | *Filtered* modifiers | Match? | Shift level |
+| ------------------ | -------------------- | ------ | ----------- |
+| (none)             | (none)               | Yes    | 1           |
+| `Shift`            | `Shift`              | Yes    | 2           |
+| `Lock`             | `Lock`               | Yes    | 2           |
+| `Shift + Lock`     | `Shift + Lock`       | No     | 1           |
+            @endfigcaption
+        </div>
+    </div>
+</div>
+
+##### Four levels
+
+The following examples compare basic types with *four levels*:
+[`FOUR_LEVEL`][FOUR_LEVEL],
+[`FOUR_LEVEL_SEMIALPHABETIC`][FOUR_LEVEL_SEMIALPHABETIC] and
+[`FOUR_LEVEL_ALPHABETIC`][FOUR_LEVEL_ALPHABETIC].
+They differ on their handling of the `Lock` modifier.
+See the [next section](@ref key-type-mappings-examples)
+for an illustration with concrete layouts.
+
+[FOUR_LEVEL]:                @ref FOUR_LEVEL
+[FOUR_LEVEL_SEMIALPHABETIC]: @ref FOUR_LEVEL_SEMIALPHABETIC
+[FOUR_LEVEL_ALPHABETIC]:     @ref FOUR_LEVEL_ALPHABETIC
+
+<div class="example-container">
+    <div class="example">
+        <div class="example-inner">
+            <div class="example-title">`FOUR_LEVEL` @anchor FOUR_LEVEL</div>
+            @figure@figcaption
+            Definition code ([source][four-level-src])
+            @endfigcaption
+```c
+type "FOUR_LEVEL" {
+	modifiers = Shift + LevelThree;
+	map[None] = Level1;
+	map[Shift] = Level2;
+
+
+	map[LevelThree] = Level3;
+	map[Shift+LevelThree] = Level4;
+
+
+
+
```

Comment:
Yes, because then the code aligns when you have two examples on the same row. But I will add a comment to explain that and replace this empty lines with empty comment.